rbac/management/migrations/0044_workspace_workspace_unique_workspace_name_per_tenant.py

```diff
@@ -0,0 +1,37 @@
+# Generated by Django 4.2.11 on 2024-06-28 10:47
+
+from django.db import migrations, models
+import django.db.models.deletion
+import django.utils.timezone
+import management.rbac_fields
+import uuid
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("api", "0014_auto_20220726_1743"),
+        ("management", "0043_auditlog"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="Workspace",
+            fields=[
+                ("id", models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name="ID")),
+                ("name", models.CharField(max_length=100)),
+                ("uuid", models.UUIDField(default=uuid.uuid4, editable=False, unique=True)),
+                ("description", models.TextField(null=True)),
+                ("created", models.DateTimeField(default=django.utils.timezone.now)),
+                ("modified", management.rbac_fields.AutoDateTimeField(default=django.utils.timezone.now)),
+                ("tenant", models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to="api.tenant")),
+            ],
+            options={
+                "ordering": ["name", "modified"],
+            },
+        ),
+        migrations.AddConstraint(
+            model_name="workspace",
+            constraint=models.UniqueConstraint(fields=("name", "tenant"), name="unique workspace name per tenant"),
```

Comment:
I think this may be too strict. We may want to allow workspaces be named the same _within_ a parent workspace. This might be one reason to store some amount of hierarchy in RBAC data. Imagine trying to create a "Production" workspace within the workspace named after a project, except someone else already created a "Production."

If we had a "parent" field, this would be unique on name and the parent foreign key.

For future consideration, this would also mean moving a Workspace could fail (i.e. 400 error) if the name conflicted with one already underneath the new parent.

Reply:
I brought this up on our UXD call today and we'll discuss it more Wednesday so we can come back to this.

Reply:
Yes you are right we should consider UXD perspective -  let's wait here for that discussion in this case.


NOTE: IG have this constraint as well [here](https://github.com/RedHatInsights/insights-host-inventory/blob/master/migrations/versions/ada40da60e4e_add_groups_and_hosts_groups_tables.py#L30)

Reply:
Yeah it does feel like we'd need it to be unique based on the parent/tenant. HBI groups don't currently support hierarchy so it makes sense that they'd have it enforced at that level.

Reply:
parent column added - uniqueness is based on combination: (name, parent, tenant)